src/util.h

```diff
@@ -28,8 +28,48 @@
 #include <string>
 #include <vector>
 
+#ifndef ANDROID
 #include <boost/signals2/signal.hpp>
-#include <boost/thread/exceptions.hpp>
+#include <boost/thread.hpp>
+
+/** Signals for translation. */
+class CTranslationInterface
+{
+public:
+    /** Translate a message to the native language of the user. */
+    boost::signals2::signal<std::string(const char *psz)> Translate;
+};
+extern CTranslationInterface translationInterface;
+
+/**
+ * Translation function: Call Translate signal on UI interface, which returns a boost::optional result.
+ * If no translation slot is registered, nothing is returned, and simply return the input.
+ */
+inline std::string _(const char *psz)
+{
+    boost::optional<std::string> rv = translationInterface.Translate(psz);
+    return rv ? (*rv) : psz;
+}
+
+#else
+
+class CTranslationInterface
+{
+public:
+    std::string Translate(const char *psz);
+};
+extern CTranslationInterface translationInterface;
+
+inline std::string _(const char *psz)
+{
+    std::string rv = translationInterface.Translate(psz);
+    if (rv.empty())
+        return std::string(psz);
+    return rv;
+}
+#endif
+
+// #include <boost/thread/exceptions.hpp>
```

Comment:
why `exceptions.hpp` has been commented?

Reply:
its not needed, I will remove

Reply:
done